packages/notification/channel/index.ts

```diff
@@ -89,42 +89,45 @@ export type NotificationMessage = {
     | NotificationStatusUpdateMessageMeta
 }
 
-type NotificationChannel<T = any> = {
-  validator: AnySchema
-  send: (notificationData: T, message: NotificationMessage) => Promise<void>
+type NotificationChannel<T = object> = {
+  validator: ArraySchema | ObjectSchema
+  send: (
+    notificationData: T | T[] | undefined,
+    message: NotificationMessage
+  ) => Promise<void>
   sendWithCustomContent?: (
-    notificationData: T,
-    customContent: T
+    notificationData: T | T[],
+    customContent: T | T[]
   ) => Promise<void>
   additionalStartupMessage?: (notificationData: T) => string
 }
```

Comment:
**Suggestion:** The NotificationChannel type has been changed to use more specific types for its properties. However, the 'sendWithCustomContent' function still uses a generic type 'T' for its parameters. It would be better to use a more specific type for these parameters to improve type safety.
```suggestion
type NotificationChannel<T = object, U = object> = {
  validator: ArraySchema | ObjectSchema
  send: (
    notificationData: T | T[] | undefined,
    message: NotificationMessage
  ) => Promise<void>
  sendWithCustomContent?: (
    notificationData: T | T[],
    customContent: U | U[]
  ) => Promise<void>
  additionalStartupMessage?: (notificationData: T) => string
}
```

Reply:
updated